sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -13342,4 +13340,34 @@ public void testCommonVirtualExpressionWithDifferentValueType() throws Exception
         ImmutableList.of()
     );
   }
+
+  // When optimization in Grouping#applyProject is applied, and it reduces a Group By query to a timeseries, we
+  // want it to return empty bucket if no row matches
+  @Test
+  public void testReturnEmptyRowWhenGroupByIsConvertedToTimeseries() throws Exception
+  {
+    skipVectorize();
+    testQuery(
+        "SELECT 'A', dim1 from foo WHERE m1 = 50 AND dim1 = 'wat' GROUP BY dim1",
```

Comment:
btw I noticed that it runs as a group-by query if we remove the `m1 = 50` clause. do you know why?

Reply:
When removing the `m1=50` clause, the `dim1` is not getting reduced to `wat` literal in the Calcite planner phase, so the optimization in `Grouping.java` to eliminate the literals is not getting applied. 
I checked the place where this reduction is happening, and it's in the `ProjectMergeRule` of Calcite. When there's a single project, then `ProjectMergeRule` is not getting invoked, which is causing this reduction to literal.